src/commands/typescript/update.ts

```diff
@@ -32,30 +37,69 @@ export default class Update extends SfdxCommand {
     }),
   };
 
-  private typescriptPkg: NpmPackage;
+  private typescriptPkgs: NpmPackage[];
 
   public async run(): Promise<void> {
-    this.typescriptPkg = this.retrieveTsPackage();
+    this.typescriptPkgs = await this.retrieveTsPackages();
     this.validateEsTarget();
     this.validateTsVersion();
 
+    const pkg = this.isLernaRepo()
+      ? await LernaRepo.create({ ux: this.ux })
+      : await SinglePackageRepo.create({ ux: this.ux });
+
     this.ux.warn('This is for testing new versions only. To update the version you must go through dev-scripts.');
+
     await this.updateTsVersion();
     await this.updateEsTarget();
 
-    const pkg = await SinglePackageRepo.create({ ux: this.ux });
-    try {
-      pkg.install();
-      pkg.build();
-      pkg.test();
-    } finally {
-      this.ux.log('Reverting unstaged stages');
-      pkg.revertUnstagedChanges();
+    if (this.isLernaRepo()) {
+      const packagePaths = await this.getPackagePaths();
+      const workingDir = pwd().stdout;
+
+      for (const packagePath of packagePaths) {
+        cd(path.resolve(packagePath));
+        try {
+          pkg.install();
+          pkg.build();
+          pkg.test();
+        } finally {
+          this.ux.log('Reverting unstaged stages');
+          cd(workingDir);
+          pkg.revertUnstagedChanges();
+        }
+        cd(workingDir);
+      }
+    } else {
+      try {
+        pkg.install();
+        pkg.build();
+        pkg.test();
+      } finally {
+        this.ux.log('Reverting unstaged stages');
+        pkg.revertUnstagedChanges();
+      }
     }
   }
 
-  private async updateEsTarget(): Promise<void> {
-    const tsConfigPath = path.resolve('tsconfig.json');
+  private isLernaRepo(): boolean {
```

Comment:
You should use this instead: https://github.com/salesforcecli/plugin-release-management/blob/main/src/repository.ts#L60